test/integration/controller/jobset_controller_test.go

```diff
@@ -643,6 +644,34 @@ func matchJobsSuspendState(js *jobset.JobSet, suspend bool) (bool, error) {
 	return true, nil
 }
 
+func matchJobsNodeSelectors(js *jobset.JobSet, nodeSelectors map[string]map[string]string) (bool, error) {
+	var jobList batchv1.JobList
+	if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
+		return false, err
+	}
+	// Count number of changed jobs
+	jobsChanged := 0
+	for _, job := range jobList.Items {
+		nodeSelectorKey, ok := job.Labels[jobset.ReplicatedJobNameKey]
+		if !ok {
+			return false, fmt.Errorf(fmt.Sprintf("%s job missing ReplicatedJobName label", job.Name))
+		}
+		if apiequality.Semantic.DeepEqual(job.Spec.Template.Spec.NodeSelector, nodeSelectors[nodeSelectorKey]) {
+			jobsChanged++
+			continue
+		}
+		return false, nil
```

Comment:
Question, why would we return early here if one of the jobs has not changed? Don't we still want to compare `jobsChanged` with `wantJobsChanged` below?

Reply:
I think It can be removed.

Reply:
we are trying to check if the child jobs got the updated nodeSelector (which is set in nodeSelectors)

@mcariatm can you switch the condition like this:

```
if !apiequality.Semantic.DeepEqual(job.Spec.Template.Spec.NodeSelector, nodeSelectors[nodeSelectorKey]) {
  return false, nil
}
jobsChanged++
```

Reply:
also, please use substitute the word `change` with `update` in all places in this function